### PR TITLE
updated README.md

### PR DESCRIPTION
**Pull request by vrastil**
updated Known installation issues

### CODE REVIEW

**Comment by elisachisari** [comment]
@vrastil: would you be up for a general update of the README? It seems some sections are now outdated. We could use this same PR for that.

**Comment by vrastil** [comment]
@elisachisari yes, I will look into that during this week

**Comment by vrastil** [comment]
README.md updated
 * unification of formatting
 * installation (issues) updated
 * functions arguments updated, mainly `int * status`
 * added density parameters function
 * example code updated

ccl_sample_run.c updated:
 * added density parameters function

File: README.md [hunk 1]
```diff
@@ -99,149 +144,169 @@ This Dockerfile currently contains all installed C libraries and the Python wrap
...
+
+For some specific cosmologies you can also use functions **`ccl_parameters_create_flat_lcdm`**, **`ccl_parameters_create_flat_wcdm`**, **`ccl_parameters_create_flat_wacdm`**, **`ccl_parameters_create_lcdm`**, which automatically set some parameters. For more information, see file ***include/ccl_core.c***.
 
-For some specific cosmologies you can also use functions **`ccl_parameters_create_flat_lcdm`**, **`ccl_parameters_create_flat_wcdm`**, **`ccl_parameters_create_flat_wacdm`**, **`ccl_parameters_create_lcdm`**, which automatically set some parameters. For more information, see file `include/ccl_core.c`.
+The status flag `int status = 0` is passed around in almost every `CCL` function. Normally zero is returned while nonzero if there were some errors during a function call. For specific cases see documentation for **`ccl_error.c`**.
 
 ### The `ccl_cosmology` object
-For the majority of CCL's functions you need an object of type **`ccl_cosmology`**, which can be initalize by function **`ccl_cosmology_create`**
+For the majority of `CCL`'s functions you need an object of type **`ccl_cosmology`**, which can be initalize by function **`ccl_cosmology_create`**
```
Comment (elisachisari) [inline README.md]: typo: initialized

File: README.md [hunk 2]
```diff
@@ -99,149 +144,169 @@ This Dockerfile currently contains all installed C libraries and the Python wrap
...
 ```c
 ccl_cosmology * ccl_cosmology_create(ccl_parameters params, ccl_configuration config);
 ```
-Note that the function returns a pointer. Variable **`params`** of type **`ccl_parameters`** contains cosmological parameters created in previous step. Structure **`ccl_configuration`** contains information about methods for computing transfer function, matter power spectrum and mass function (for available methods see `include/ccl_config.h`). For now, you should use default configuration **`default_config`**
+Note that the function returns a pointer. Variable `params` of type **`ccl_parameters`** contains cosmological parameters created in previous step. Structure **`ccl_configuration`** contains information about methods for computing transfer function, matter power spectrum and mass function (for available methods see ***include/ccl_config.h***). For now, you should use default configuration `default_config`
```
Comment (elisachisari) [inline README.md]: I would remove "For now". We do have other capabilities. I would rephrase this sentence as: "In the default configuration, CCL will use the following set-up:"

File: README.md [hunk 3]
```diff
@@ -99,149 +144,169 @@ This Dockerfile currently contains all installed C libraries and the Python wrap
...
 				int has_rsd,int has_magnification,int has_intrinsic_alignment,
 				int nz_n,double *z_n,double *n,
 				int nz_b,double *z_b,double *b,
 				int nz_s,double *z_s,double *s,
 				int nz_ba,double *z_ba,double *ba,
-				int nz_rf,double *z_rf,double *rf);
+				int nz_rf,double *z_rf,double *rf, int * status);
 ````
-Exact definition of these parameters are described in file *include/ccl_cls.h*. Usually you can use simplified versions of this function, namely **ccl_cl_tracer_number_counts_new, ccl_cl_tracer_number_counts_simple_new, ccl_cl_tracer_lensing_new** or **ccl_cl_tracer_lensing_simple_new**. Two most simplified versions (one for number counts and one for shear) take parameters:
+Exact definition of these parameters are described in file ***include/ccl_cls.h***. Usually you can use simplified versions of this function, namely **`ccl_cl_tracer_number_counts_new`, `ccl_cl_tracer_number_counts_simple_new`, `ccl_cl_tracer_lensing_new`** or **`ccl_cl_tracer_lensing_simple_new`**. Two most simplified versions (one for number counts and one for shear) take parameters:
```
Comment (elisachisari) [inline README.md]: More of a comment for the code than for the README: why do we have "new" at the name of each function name? I think we should rename these.

File: README.md [hunk 4]
```diff
@@ -99,149 +144,169 @@ This Dockerfile currently contains all installed C libraries and the Python wrap
...
 
 ### Halo mass function
-The halo mass function *dN/dM* can be obtained by function **ccl_massfunc**
+The halo mass function *dN/dM* can be obtained by function **`ccl_massfunc`**
 ````c
-double ccl_massfunc(ccl_cosmology * cosmo, double halo_mass, double redshift)
+double ccl_massfunc(ccl_cosmology * cosmo, double smooth_mass, double a, double odelta, int * status);
 ````
-where **halo_mass** is mass smoothing scale (in units of *M_sun/h*. For more details (or other functions like *sigma_M*) see *include/ccl_massfunc.h* and *src/mass_func.c*.
+where `smooth_mass` is mass smoothing scale (in units of *M_sun*) and `odelta` is choice of Delta. For more details (or other functions like **`sigma_M`**) see ***include/ccl_massfunc.h*** and ***src/mass_func.c***.
```
Comment (elisachisari) [inline README.md]: Units have changed? @villarrealas can you verify this?

File: README.md [hunk 5]
```diff
@@ -99,149 +144,169 @@ This Dockerfile currently contains all installed C libraries and the Python wrap
...
 
 ### LSST Specifications
-CCL includes LSST specifications for the expected galaxy distributions of the full galaxy clustering sample and the lensing source galaxy sample. Start by defining a flexible photometric redshift model given by function
+`CCL` includes LSST specifications for the expected galaxy distributions of the full galaxy clustering sample and the lensing source galaxy sample. Start by defining a flexible photometric redshift model given by function
 ````c
-double (* your_pz_func)(double photo_z, double spec_z, void *param);
+double (* your_pz_func)(double z_ph, double z_spec, void *param, int * status);
 ````
-which returns the probability of measuring a particular photometric redshift, given a spectroscopic redshift and other relevant parameters. Then you call function **ccl_specs_create_photoz_info**
+which returns the liklihood of measuring a particular photometric redshift `z_ph` given a spectroscopic redshift `z_spec`, with a pointer to additional arguments `param` and a status flag. Then you call function **`ccl_specs_create_photoz_info`**
```
Comment (elisachisari) [inline README.md]: typo: likelihood

File: README.md [hunk 6]
```diff
@@ -99,149 +144,169 @@ This Dockerfile currently contains all installed C libraries and the Python wrap
...
 typedef struct {
-	//first double corresponds to photo-z, second to spec-z
-        double (* your_pz_func)(double, double, void *);
-        void *  your_pz_params;
+  double (* your_pz_func)(double, double, void *, int*);
+  void *  your_pz_params;
 } user_pz_info;
 ````
-The expected *dN/dz* for lensing or clustering galaxies with given binnig can be obtained by function **ccl_specs_dNdz_tomog**
+The expected *dN/dz* for lensing or clustering galaxies with given binnig can be obtained by function **`ccl_specs_dNdz_tomog`**
```
Comment (elisachisari) [inline README.md]: typo: binning

File: README.md [hunk 7]
```diff
@@ -258,165 +323,208 @@ where */path/to/install/* is the path to the location where the library has been
...
+A Python wrapper for `CCL` is provided through a module called `pyccl`. The whole `CCL` interface can be accessed through regular Python functions and classes, with all of the computation happening in the background through the C code. The functions all support `numpy` arrays as inputs and outputs, with any loops being performed in the C code for speed.
 
-The Python module has essentially the same functions as the C library, just presented in a more standard Python-like way. You can inspect the available functions and their arguments by using the built-in Python **help()** function, as with any Python module.
+The Python module has essentially the same functions as the C library, just presented in a more standard Python-like way. You can inspect the available functions and their arguments by using the built-in Python **`help()`** function, as with any Python module.
```
Comment (elisachisari) [inline README.md]: This seems to be repeated information

File: README.md (hunk 3 above)
Comment (vrastil) [reply README.md]: I think that use of **`new`** should be a hint for the user to clean up after he is finished with some sort of **`free`**

File: README.md (hunk 4 above)
Comment (vrastil) [reply README.md]: I changed this based on information in ***include/ccl_massfunc.h***

Comment (villarrealas) [reply README.md]: That is correct - I must have missed that I hadn't changed things in the readme. ccl_massfunc uses units of M_sun consistently throughout.

File: README.md (hunk 3 above)
Comment (elisachisari) [reply README.md]: Ok, I'll think whether we want to change this eventually
